tests/src/esmvaltool/test_search.py

```diff
@@ -0,0 +1,139 @@
+"""Test search module."""
+import gzip
+import json
+from pathlib import Path
+from unittest import mock
+
+import pytest
+from esmvalcore.config import CFG
+from esmvalcore.dataset import Dataset
+
+from ewatercycle.esmvaltool import search
+
+DATASET_TEMPLATE = {
+    "short_name": "pr",
+    "activity": "ScenarioMIP",
+    "mip": "E1hr",
+    "project": "CMIP6",
+    "exp": "ssp585",
+    "dataset": "CNRM-CM6-1-HR",
+    "institute": "CNRM-CERFACS",
+    "ensemble": "r1i1p1f2",
+    "grid": "gr",
+}
+CFG["search_esgf"] = "always"
+
+STORED_RESULTS = Path(__file__).parent / "files" / "query_result.txt.gz"
+
+
+@pytest.fixture(scope="session")
+def mock_esgf_query():
+    """Mocks the ESGF query to allow for testing without accessing their servers.
+
+    Returns the result of running the following code:
+
+    .. code-block:: python
+
+        query_result = _query_esgf(
+            activity="ScenarioMIP",
+            experiment="ssp585",
+            variables=["pr", "tas", "rsds"],
+        )
+
+    You should be able to recreate this file with:
+
+    .. code-block:: python
+
+        import gzip
+
+        serialized = "\n".join(
+            str(query).replace("\n", "").replace("'",'"')
+            for query in query_result
+        )
+        out = serialized.replace("Dataset:", "")
+
+        with gzip.open('query_result.txt.gz', 'wb') as f:
+            f.write(out.encode())
```

Comment:
Using output of the Dataset.__repr__() is bit a cumbersome.
The repr prints self.facets dict, which is serializable. 
So this could be written as

```suggestion
        out = json.dumps([d.facets for d in query_result])
        with gzip.open('query_result.txt.gz', 'wb') as f:
            f.write(out.encode())
```

For the mock you would read the file, parse json and pass each list item to the Dataset constructor.